From a79755a7099bb059ab0424e8e31c7da2338a1098 Mon Sep 17 00:00:00 2001
From: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
Date: Sun, 14 Nov 2021 19:20:44 +1100
Subject: [PATCH 03/15] msdasql: Implement IColumnsRowset GetAvailableColumns

The functions returns a series of custom columns to make a row unique.
For us, it's safe to return no extra columns at this stage.

Signed-off-by: Alistair Leslie-Hughes <leslie_alistair@hotmail.com>
---
 dlls/msdasql/session.c | 12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

diff --git a/dlls/msdasql/session.c b/dlls/msdasql/session.c
index 533cb6ae28a..98b2b4e6c5d 100644
--- a/dlls/msdasql/session.c
+++ b/dlls/msdasql/session.c
@@ -886,8 +886,16 @@ static ULONG WINAPI column_rs_Release(IColumnsRowset *iface)
 static HRESULT WINAPI column_rs_GetAvailableColumns(IColumnsRowset *iface, DBORDINAL *count, DBID **columns)
 {
     struct msdasql_rowset *rowset = impl_from_IColumnsRowset( iface );
-    FIXME("%p, %p, %p\n", rowset, count, columns);
-    return E_NOTIMPL;
+
+    TRACE("%p, %p, %p\n", rowset, count, columns);
+
+    if (!count || !columns)
+        return E_INVALIDARG;
+
+    *count = 0;
+    *columns = NULL;
+
+    return S_OK;
 }
 
 static HRESULT WINAPI column_rs_GetColumnsRowset(IColumnsRowset *iface, IUnknown *outer, DBORDINAL count,
-- 
2.33.0

